packages/rum-core/src/domain/rumEventsCollection/resource/resourceCollection.ts

```diff
@@ -52,6 +60,22 @@ export function startResourceCollection(
   })
 }
 
+function waitForResponseToFinish(request: RequestCompleteEvent, callback: (duration: Duration) => void) {
+  const duration = request.duration || elapsed(request.startClocks.timeStamp, timeStampNow())
```

Comment:
for `XHR` request.duration should be defined, else we should create one as a fallback